### DISCUSSION TITLE
Sometimes EncryptionServiceProvider not getting configs form .env "No application encryption key has been specified" error!

### DISCUSSION THREAD

**Discussion by ss-koshala** [laravel/nova-issues#4976] [Q&A]
- Laravel Version: 9.37.0
- Nova Version: 4.17.1
- PHP Version: 8.0.13
- Database Driver & Version: MySQL 5.7.36
- Operating System and Version: Windows 11
- Browser type and version: Chrome
- Reproduction Repository: https://github.com/###/###

### Description:
Sometimes EncryptionServiceProvider not getting configs form .env
$config = $app->make('config')->get('app');
here $config not getting from .env file. So key filed empty and getting "No application encryption key has been specified" error.

### Detailed steps to reproduce the issue on a fresh Nova installation:

In laravel nova admin load "View" on a resource, that resource should have relation fields like, HasOne, or HasMany or any other relation field. And then go to the "view". like this "admin/resources/members/1#Documents=nda" and reload multiple times, we can see console errors.

nova-api/hotel-reservations?search=&filters=W10%3D&orderBy=&perPage=5&trashed=&page=1&viaResource=members&viaResourceId=1&viaRelationship=hotelReservations&relationshipType=hasMany

"nova-api/roles/actions?viaResource=users&viaResourceId=2&viaRelationship=roles&relationshipType=morphToMany&display=index"
like this.

**Comment by crynobone** [comment]
Nova doesn't interfere with either `.env` or `EncryptionServiceProvider`. Therefore, this should probably be configuration issues or reported to `laravel/framework`.

**Reply by ss-koshala** [reply]
it does not a matter env or EncryptionServiceProvider error only comming when use nova relation fields. 

HasOne::make('Travel Datum', 'TravelDatum'),
HasMany::make('Hotel Reservations','hotelReservations', HotelReservation::class),
BelongsToMany::make('Hotels', 'hotels', Hotel::class),

If I didn'nt use these fields I'm not getting these errors.

**Reply by crynobone** [reply]
> If I didn'nt use these fields I'm not getting these errors.

None of these fields interact with "encryption"

-------

Can you show me the full error log from `laravel.log`?

**Reply by ss-koshala** [reply]
`[2022-10-27 11:33:33] production.ERROR: No application encryption key has been specified. {"exception":"[object] (Illuminate\\Encryption\\MissingAppKeyException(code: 0): No application encryption key has been specified. at C:\\wamp\\www\\lmc.gueststar\\vendor\\laravel\\framework\\src\\Illuminate\\Encryption\\EncryptionServiceProvider.php:79)
[stacktrace]
#0 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Support\\helpers.php(306): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}(NULL)
#1 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Encryption\\EncryptionServiceProvider.php(81): tap(NULL, Object(Closure))
#2 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Encryption\\EncryptionServiceProvider.php(60): Illuminate\\Encryption\\EncryptionServiceProvider->key(Array)
#3 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Encryption\\EncryptionServiceProvider.php(32): Illuminate\\Encryption\\EncryptionServiceProvider->parseKey(Array)
#4 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(871): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}(Object(Illuminate\\Foundation\\Application), Array)
#5 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(756): Illuminate\\Container\\Container->build(Object(Closure))
#6 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(860): Illuminate\\Container\\Container->resolve('encrypter', Array, true)
#7 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(692): Illuminate\\Foundation\\Application->resolve('encrypter', Array)
#8 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(845): Illuminate\\Container\\Container->make('encrypter', Array)
#9 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(1029): Illuminate\\Foundation\\Application->make('encrypter')
#10 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(945): Illuminate\\Container\\Container->resolveClass(Object(ReflectionParameter))
#11 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(906): Illuminate\\Container\\Container->resolveDependencies(Array)
#12 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(756): Illuminate\\Container\\Container->build('App\\\\Http\\\\Middle...')
#13 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(860): Illuminate\\Container\\Container->resolve('App\\\\Http\\\\Middle...', Array, true)
#14 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(692): Illuminate\\Foundation\\Application->resolve('App\\\\Http\\\\Middle...', Array)
#15 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(845): Illuminate\\Container\\Container->make('App\\\\Http\\\\Middle...', Array)
#16 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(169): Illuminate\\Foundation\\Application->make('App\\\\Http\\\\Middle...')
#17 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#18 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(726): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#19 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(703): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#20 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(667): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#21 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(656): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#22 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(190): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#23 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#24 C:\\wamp\\www\\larasite\\vendor\\laravel\\nova\\src\\Http\\Middleware\\ServeNova.php(23): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#25 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Laravel\\Nova\\Http\\Middleware\\ServeNova->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#26 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#27 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#28 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#29 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#30 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#31 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#32 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#33 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#34 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#35 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#36 C:\\wamp\\www\\larasite\\vendor\\fruitcake\\laravel-cors\\src\\HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#37 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Fruitcake\\Cors\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#38 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Http\\Middleware\\TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#39 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#40 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#41 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(165): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#42 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(134): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#43 C:\\wamp\\www\\larasite\\public\\index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#44 {main}
"} `

**Reply by ss-koshala** [reply]
---Special note---

This error not happend always, only resource view some times getting the error.

**Reply by crynobone** [reply]
```
#9 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(1029): Illuminate\\Foundation\\Application->make('encrypter') 
#10 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(945): Illuminate\\Container\\Container->resolveClass(Object(ReflectionParameter))

#12 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(756): Illuminate\\Container\\Container->build('App\\\\Http\\\\Middle...') 
#13 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(860): Illuminate\\Container\\Container->resolve('App\\\\Http\\\\Middle...', Array, true) 
#14 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(692): Illuminate\\Foundation\\Application->resolve('App\\\\Http\\\\Middle...', Array) 
#15 C:\\wamp\\www\\larasite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(845): Illuminate\\Container\\Container->make('App\\\\Http\\\\Middle...', Array)
```

The error originated from `App\Http\Middle...`, this is outside of Nova and we have no control over this.

Again as I said earlier. "this should probably be configuration issues or reported to laravel/framework". Feel free to thumb down but no one else reporting this issue so you should really consider a usage issue on your application.
